File: algorithms/eulerTotientFunctionPattern/eulerFunction.c
#include <stdio.h>

int gcd(int a, int b) 
{ 
    if (a == 0) 
        return b; 
    return gcd(b % a, a); 
}
int phi(unsigned int n) 
{ 
    unsigned int result = 1; 
    for (int i = 2; i < n; i++) 
        if (gcd(i, n) == 1) 
            result++; 
    return result; 
}
int main() 
{ 
    int n; 
    for (n = 1; n <= 10; n++) 
        printf("phi(%d) = %d\n", n, phi(n)); 
    return 0; 
} 

File: algorithms/eulerTotientFunctionPattern/eulerFunction.java
import java.io.*; 
  
public class eulerFunction {
    public static int gcd(int a, int b) 
    { 
        if (a == 0) 
            return b; 
        return gcd(b % a, a); 
    } 
    public static int phi(int n) 
    { 
        int result = 1; 
        for (int i = 2; i < n; i++) 
            if (gcd(i, n) == 1) 
                result++; 
        return result; 
    } 
    public static void main(String[] args)
    { 
        int n; 
  
        for (n = 1; n <= 20; n++) 
            System.out.println("phi(" + n + ") = " + phi(n)); 
    } 
}